### ISSUE TITLE
Proposal: Add new BufferAttribute methods

### ISSUE THREAD

**Issue by WestLangley** [mrdoob/three.js#18006]
Vector3 has these methods:
```js
Vector3.applyMatrix3( matrix3 );
Vector3.applyNormalMatrix( matrix3 );
Vector3.applyMatrix4( matrix4 );
Vector3.transformDirection( matrix4 );
```

We need to be able to apply the same transforms to each element of a `BufferAttribute`.

I propose we add the following methods:
```js
BufferAtribute.applyMatrix3( matrix3 );       // to replace Matrix3.applyToBufferAttribute()
BufferAtribute.applyNormalMatrix( matrix3 );  // new
BufferAtribute.applyMatrix4( matrix4 );       // to replace Matrix4.applyToBufferAttribute()
BufferAtribute.transformDirection( matrix4 ); // new
```

Related:
#17997
#18001 bug
#18005 bug

Suggestions for alternate APIs welcome.

**Comment by mrdoob** [comment]
`transformDirection()` > `applyDirectionMatrix()`?

**Comment by WestLangley** [comment]
>transformDirection() > applyDirectionMatrix()

`transformDirection()` is applying a transform to the vector, but interpreting the vector not as a position vector, but as a direction vector. That method was introduces a long time ago and I think the name is OK.

**Comment by WestLangley** [comment]
I never liked the API
```js
matrix.applyToBufferAttribute( attribute );
```
and this is an opportunity to replace it with an API that is, in my opinion, more natural.

**Comment by mrdoob** [comment]
Moving them to the attribute sounds good to me. Was just trying to unify the last name.

**Comment by Mugen87** [comment]
> Moving them to BufferAttribute sounds good to me.

👍

**Comment by WestLangley** [comment]
OK, I'll do it.

**Comment by WestLangley** [comment]
Closed in #18070.
